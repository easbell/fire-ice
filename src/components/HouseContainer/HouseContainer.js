import React, { Component } from 'react'
import { connect } from 'react-redux';
import { House } from '../House/House';
import './HouseContainer.css';

export class HouseContainer extends Component {
  renderCards = () => {
    const { houses } = this.props;
    return houses.map(house => (
      <House key={house.name} {...house} />
    ))
  }

  render() {
    return (
      <div className='container'>
        {this.renderCards()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  houses: state.houses
});

export default connect(mapStateToProps)(HouseContainer);