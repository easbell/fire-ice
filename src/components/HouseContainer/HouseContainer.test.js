import React from 'react';
import { HouseContainer, mapStateToProps } from './HouseContainer';
import { shallow } from 'enzyme';

describe('HouseContainer', () => {
  describe('mapStateToProps', () => {
    it('should return an object with loading', () => {
      const mockState = {
        houses: ['house', 'house'],
        loading: true,
        other: 'other'
      }
      const expected = {houses: ['house', 'house']}
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected)
    });
  });
});