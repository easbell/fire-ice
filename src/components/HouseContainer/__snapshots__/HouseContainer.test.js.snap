// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HouseContainer
    houses={
      Array [
        "house",
        "house",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <House
          0="h"
          1="o"
          2="u"
          3="s"
          4="e"
        />,
        <House
          0="h"
          1="o"
          2="u"
          3="s"
          4="e"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "0": "h",
          "1": "o",
          "2": "u",
          "3": "s",
          "4": "e",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "0": "h",
          "1": "o",
          "2": "u",
          "3": "s",
          "4": "e",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <House
            0="h"
            1="o"
            2="u"
            3="s"
            4="e"
          />,
          <House
            0="h"
            1="o"
            2="u"
            3="s"
            4="e"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "0": "h",
            "1": "o",
            "2": "u",
            "3": "s",
            "4": "e",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "0": "h",
            "1": "o",
            "2": "u",
            "3": "s",
            "4": "e",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
